src/Main.elm

```diff
@@ -1,71 +1,392 @@
+            , margin auto
+            , position relative
+            ]
+        ]
+        [ viewHeader model
+        , ul []
+            (viewTodos model)
+        ]
+
 
+onKeyDown : (Int -> msg) -> Attribute msg
+onKeyDown tagger =
+    on "keydown" (Json.map tagger keyCode)
 
-viewTodo : Todo -> Html msg
-viewTodo todo =
-    li []
-        [ h1 [] [ text todo.title ]
+
+viewHeader : Model -> Html Msg
+viewHeader model =
+    header
+        [ css
+            [ textAlign center
+            ]
+        ]
+        [ h1 [] [ text "Todos" ]
+        , input
+            [ css
+                [ Css.width (pct 75)
+                ]
+            , value model.newTodoTitle
+            , onInput CHANGE_INPUT
+            , onKeyDown (KEY_DOWN OnAdd)
+            , placeholder "What needs to be done"
+            ]
+            []
+        , button [ onClick ADD_TODO ] [ text "Add" ]
+        , div []
+            [ input
+                [ onCheck HIDE_COMPLETED
+                , type_ "checkbox"
+                , HSA.checked model.hideCompleted
+                ]
+                []
+            , label [] [ text "Hide completed" ]
+            ]
         ]
 
 
-viewTodos : List Todo -> List (Html msg)
-viewTodos todos =
-    List.map viewTodo todos
+viewTodo : Model -> Todo -> Html Msg
+viewTodo model todo =
+    li
+        [ css
+            [ listStyleType none
+            , backgroundColor (rgb 240 240 240)
+            , padding (px 10)
+            , Css.nthChild "even"
+                [ backgroundColor (rgb 220 220 220)
+                ]
+            , if todo.completed then
+                textDecoration lineThrough
+
+              else
+                textDecoration none
+            ]
+        ]
+        [ div []
+            [ input
+                [ type_ "checkbox"
+                , HSA.checked todo.completed
+                , onCheck (CHANGE_COMPLETED todo)
+                ]
+                []
+            , if model.todoBeingEdited == todo.id then
+                input
+                    [ id "todo-edit-input"
+                    , value todo.title
+                    , onBlur (END_EDITTING todo)
+                    , onInput (EDIT_TODO todo)
+                    , onKeyDown (KEY_DOWN OnRename)
+                    ]
+                    []
+
+              else
+                label
+                    [ onClick (START_EDITING todo)
+                    ]
+                    [ text todo.title ]
+            , button
+                [ css
+                    [ Css.float right
+                    ]
+                , onClick (DELETE_TODO todo)
+                ]
+                [ text "x" ]
+            ]
+        ]
+
+
+viewTodos : Model -> List (Html Msg)
+viewTodos model =
+    List.map (viewTodo model)
+        (if model.hideCompleted == True then
+            model.todos
+                |> List.filter (\todo -> todo.completed == False)
+
+         else
+            model.todos
+        )
```

Comment:
Seems like you shy away from `let...in` blocks. You could compute the filtered todos in such a block.